library/std/src/io/mod.rs

```diff
@@ -2389,16 +2389,50 @@ impl<T: Read, U: Read> Read for Chain<T, U> {
         }
         self.second.read_vectored(bufs)
     }
+
+    #[inline]
+    fn is_read_vectored(&self) -> bool {
+        self.first.is_read_vectored() || self.second.is_read_vectored()
+    }
+
+    fn read_to_end(&mut self, buf: &mut Vec<u8>) -> Result<usize> {
+        let mut read = 0;
+        if !self.done_first {
+            read += self.first.read_to_end(buf)?;
+            self.done_first = true;
+        }
+        read += self.second.read_to_end(buf)?;
+        Ok(read)
+    }
```

Comment:
Hm... what if the combined read is larger than `usize`? I guess that isn't realistic to worry about, as if that happens, our Vec is going to suffer allocation failure. Makes me feel like `io::Read` is very awkward, however.

Reply:
Yes if the result is larger than `usize` there is either a bug in the other implementation or an allocation failure in the `Vec`